internal/events/events.go

```diff
@@ -152,6 +153,13 @@ func DeleteEvent(ctx context.Context, db sql.Executor, id types.UUID) error {
 	return err
 }
 
+// DeleteEvents deletes a list of events from the event store.
+// It is idempotent. If the event does not exist, it will not return an error.
+func DeleteEvents(ctx context.Context, db sql.DB, ids ...types.UUID) error {
+	_, err := db.Execute(ctx, deleteEvents, types.UUIDArray(ids))
+	return err
+}
+
 // GetUnreceivedEvents retrieves events that are neither received by the network nor previously broadcasted.
 // An event is considered "received" only after its inclusion in a block.
 // The function excludes events that have been broadcasted but are still pending in the mempool, awaiting block inclusion.
```

Comment:
I think we'd remove `GetUnreceivedEvents` now.  But I'm still digesting that the new approach would pull the event IDs that need broadcast from voting store now.